crates/sui-storage/src/event_store/sql.rs

```diff
@@ -313,9 +371,16 @@ impl From<SqliteRow> for StoredEvent {
             .expect("Error converting stored sender address bytes to SuiAddress");
         let recipient = SqlEventStore::try_extract_recipient(&row)
             .expect("Error converting stored recipient address to Owner");
-        let object_version = SqlEventStore::try_extract_object_version(&row);
-        let transfer_type = SqlEventStore::try_extract_transfer_type(&row)
-            .expect("Error converting stored transfer type to TransferType");
+
+        let fields_json_value = SqlEventStore::try_extract_extra_fields(&row)
+            .expect("Error parsed extra fields into JSON value");
+
+        let object_version =
```

Comment:
Instead of trying to populate `object_version` and `transfer_type` fields when we create StoredEvent, can we do this at the last possible moment, ie when we populate `SuiEventEnvelope`?

The reason is that doing it here in `StoredEvent` uses up extra memory (since it is already in fields).  
It is also possible that someone might use the EventStore API to read some fields and do extra filtering, or something else, and thus not need the field right away.

Reply:
those variables are optional, so it's only populated when the field is stored in DB.

Reply:
resolved off line, will pdate